HypVINN/inference.py

```diff
@@ -102,41 +168,151 @@ def set_model(self, cfg=None):
+
+        Returns
+        -------
+        str
+            The name of the model.
+        """
         return self.model_name
 
     def get_cfg(self):
+        """
+        Get the configuration node.
+
+        This method returns the configuration node used in the Inference instance.
+
+        Returns
+        -------
+        yacs.config.CfgNode
+            The configuration node containing the parameters for the model.
+        """
         return self.cfg
 
     def get_num_classes(self):
+        """
+        Get the number of classes.
+
+        This method returns the number of classes defined in the model configuration.
+
+        Returns
+        -------
+        int
+            The number of classes.
+        """
         return self.cfg.MODEL.NUM_CLASSES
 
     def get_plane(self):
+        """
+        Get the plane.
+
+        This method returns the plane defined in the data configuration.
+
+        Returns
+        -------
+        str
+            The plane.
+        """
         return self.cfg.DATA.PLANE
 
     def get_model_height(self):
+        """
+        Get the model height.
+
+        This method returns the height of the model defined in the model configuration.
+
+        Returns
+        -------
+        int
+            The height of the model.
+        """
         return self.cfg.MODEL.HEIGHT
 
     def get_model_width(self):
+        """
+        Get the model width.
+
+        This method returns the width of the model defined in the model configuration.
+
+        Returns
+        -------
+        int
+            The width of the model.
+        """
         return self.cfg.MODEL.WIDTH
 
     def get_max_size(self):
+        """
+        Get the maximum size of the output tensor.
+
+        Returns
+        -------
+        int or tuple
+            The maximum size. If the width and height of the output tensor are equal, it returns the width. Otherwise, it
+            returns both the width and height.
+        """
         if self.cfg.MODEL.OUT_TENSOR_WIDTH == self.cfg.MODEL.OUT_TENSOR_HEIGHT:
             return self.cfg.MODEL.OUT_TENSOR_WIDTH
         else:
             return self.cfg.MODEL.OUT_TENSOR_WIDTH, self.cfg.MODEL.OUT_TENSOR_HEIGHT
 
     def get_device(self):
+        """
+        Get the device.
+
+        This method returns the device and view aggregation device used in the Inference instance.
+
+        Returns
+        -------
+        tuple
+            The device and view aggregation device.
+        """
         return self.device,self.viewagg_device
 
     #TODO check is possible to modify to CerebNet inference mode from RAS directly to LIA (CerebNet.Inference._predict_single_subject)
     @torch.no_grad()
-    def eval(self, val_loader: DataLoader, pred_prob: torch.Tensor, out_scale=None):
+    def eval(self, val_loader: DataLoader, pred_prob: torch.Tensor, out_scale: float = None) -> torch.Tensor:
+        """
```

Comment:
I will remove the validation part, as this is the same function used for any dataset. 

Evaluate the model on a HypVINN dataset.
This method runs the model in evaluation mode on a Hypvinn Dataset. It iterates over the given dataset and
 computes the model's predictions.

Reply:
changed